jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -51,7 +51,17 @@ def compatible_units(data):
             if not len(self.layers):
                 return True
 
-            data_xunit = data.get_component(str(self.state.x_att)).units
+            # If we load, e.g.,  one spectrum in Frequency and one in Wavelength,
+            # the viewer state x_att won't match the component of the second spectrum since
+            # they're no longer both the non-specific "World 0"
```

Comment:
might just be useful for us to remember why this code block is here.  Is there a PR to point to as well maybe?

```suggestion
            # (as of Specutils 2.X) they're no longer both the non-specific "World 0"
```